### DISCUSSION TITLE
Size of the arena

### DISCUSSION THREAD

**Discussion by Marirapellini** [NeLy-EPFL/flygym#98] [Q&A] [answered]
Hi! I am trying to figure out how the dimensions for the Odor Arena works. It says by default the size is (50,50) mm but I do not understand why that is the case (I only see that the FlatTerrain's size is (50,50)). Could you please explain me how the size of course?
The idea is to truncate the simulation when the fly walks off the arena but since I do not understand how these sizes work I am a bit lost
Thank you!

**Comment by sibocw** [comment, marked as answer]
Hi,

Taking `FlatTerrain` for example, the ground is actually initialized [here](https://github.com/NeLy-EPFL/flygym/blob/25b7dbf01230cdef6fe8c521e08612b5abf55d35/flygym/mujoco/arena/base.py#L212-L219) as a MuJoCo plane:
```Python
        ground_size = [*size, 1]

        ...

        self.root_element.worldbody.add(
            "geom",
            type="plane",
            name="ground",
            material=grid,
            size=ground_size,
            friction=friction,
        )
```

As you can see the `ground_size` is provided as the `size` parameter for the MuJoCo. Based on the [MuJoCo docs](https://mujoco.readthedocs.io/en/latest/XMLreference.html#body-geom) on this, the `size` is "X half-size; Y half-size; spacing between square grid lines for rendering. If either the X or Y half-size is 0, the plane is rendered as infinite in the dimension(s) with 0 size."

In the case of the `OdorArena`, as we discussed, the default size specified in the documentation is wrong. It should be (300, 300) instead of (50, 50). Sorry about that.

**Comment by sibocw** [comment]
Fixed in (v0.2.4)[https://github.com/NeLy-EPFL/flygym/releases/tag/v0.2.4]
